src/sos_utils.py

```diff
@@ -0,0 +1,77 @@
+"""Collections of sos related utilities."""
+import glob
+import logging
+from pathlib import Path
+from subprocess import DEVNULL, CalledProcessError, check_call
+from typing import List, Optional, Union
+
+from uploader import Uploader
+
+logger = logging.getLogger(__name__)
+
+
+class ReportNotFoundError(Exception):
+    """Report not found exception."""
+
+
+class SoSCollectHelper:
+    """Helper class for 'sos collect'."""
+
+    sos_collect_cmd_base = "sos collect"
+    sos_default_options = "--no-local --nopasswd-sudo --batch"
```

Comment:
**nitpick:** 

Maybe list of string can bring us more flexible?

Reply:
This one is not blocking, I just think if it's a list then we can more easy to handle the options without hardcoded string formatting. Like


```python
options = ["--no-local", "--nopasswd-sudo", "--batch"]
new_options: List[str] = ["--abc", "--qwe"]

cmd = "sos {}".format(" ".join(options+new_options))
```